neqo-transport/src/recv_stream.rs

```diff
@@ -360,6 +355,23 @@ impl RecvStream {
         }
     }
 
+    fn set_state(&mut self, new_state: RecvStreamState) {
+        qtrace!(
+            "RecvStream {} state {} -> {}",
+            self.stream_id.as_u64(),
+            self.state.name(),
+            new_state.name()
+        );
+
+        if let RecvStreamState::Recv { .. } = &self.state {
```

Comment:
Should you be checking that the discriminator is the same and not doing anything if things haven't changed?  It seems like this might clear the MAX_STREAM_DATA too often (and it's not a free operation).

Reply:
The code as-is will not. `set_state` is only called in cases where we know the state and are changing to a known different state. I'm going to add an assert for this, just in case.

Also, `Recv` state is only exited once per stream so there's no potential for thrashing. Stating this requires knowledge of ReadStream, rather than something enforced by FlowMgr. (It would be nice if FlowMgr inherently did the right thing but that will require a bigger FlowMgr re-think.)